Sources/RoutingKit/TrieRouter/TrieRouter.swift

```diff
@@ -0,0 +1,98 @@
+import Logging
+import Synchronization
+
+/// Generic ``TrieRouter`` built using the "trie" tree algorithm.
+///
+/// Use ``register(_:at:)`` to register routes into the router. Use ``route(path:parameters:)`` to then fetch
+/// a matching route's output.
+///
+/// See https://en.wikipedia.org/wiki/Trie for more information.
+public final class TrieRouter<Output: Sendable>: Router, Sendable, CustomStringConvertible {
+    typealias Node = TrieRouterNode<Output>
+
+    /// Available ``TrieRouter`` customization options.
+    public enum ConfigurationOption: Sendable {
+        /// If set, this will cause the router's route matching to be case-insensitive.
+        ///
+        /// > Note: Case-insensitive routing may be less performant than case-sensitive routing.
+        case caseInsensitive
+    }
+
+    /// The root node.
```

Comment:
```suggestion
```
This comment is kinda redundant, especially since private symbols never get rendered in docs.